scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateIfTaskHandler.java

```diff
@@ -0,0 +1,190 @@
+package org.ow2.proactive.scheduler.job.termination.handlers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.NotificationData;
+import org.ow2.proactive.scheduler.common.SchedulerEvent;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+import org.ow2.proactive.scheduler.common.task.flow.FlowAction;
+import org.ow2.proactive.scheduler.core.SchedulerStateUpdate;
+import org.ow2.proactive.scheduler.job.ChangedTasksInfo;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.job.JobInfoImpl;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+
+public class TerminateIfTaskHandler {
+
+	public static final Logger logger = Logger.getLogger(TerminateIfTaskHandler.class);
+
+	private final InternalJob internalJob;
+
+	public TerminateIfTaskHandler(InternalJob internalJob) {
+		this.internalJob = internalJob;
+	}
+
+	public boolean terminateIfTask(FlowAction action, InternalTask initiator, ChangedTasksInfo changesInfo,
+			SchedulerStateUpdate frontend, InternalTask descriptor, TaskId taskId) {
+
+		InternalTask[] targets = searchIfElseJoinTasks(action, initiator);
+
+		// the targetIf from action.getTarget() is the selected branch;
+		// the IF condition has already been evaluated prior to being put in a
+		// FlowAction
+		// the targetElse from action.getTargetElse() is the branch that was NOT
+		// selected
+		InternalTask targetIf = targets[0];
+		InternalTask targetElse = targets[1];
+		InternalTask targetJoin = targets[2];
+
+		logger.info("Control Flow Action IF: " + targetIf.getId() + " join: "
+				+ ((targetJoin == null) ? "null" : targetJoin.getId()));
+
+		// these 2 tasks delimit the Task Block formed by the IF branch
+		InternalTask branchStart = targetIf;
+		InternalTask branchEnd = null;
+
+		String match = targetIf.getMatchingBlock();
+		if (match != null) {
+			for (InternalTask t : internalJob.getIHMTasks().values()) {
+				if (match.equals(t.getName())
+						&& !(t.getStatus().equals(TaskStatus.FINISHED) || t.getStatus().equals(TaskStatus.SKIPPED))) {
+					branchEnd = t;
+				}
+			}
+		}
+		// no matching block: there is no block, the branch is a single task
+		if (branchEnd == null) {
+			branchEnd = targetIf;
+		}
+
+		// plug the branch
+		branchStart.addDependence(initiator);
+		changesInfo.taskUpdated(branchStart);
+		if (targetJoin != null) {
+			targetJoin.addDependence(branchEnd);
+			changesInfo.taskUpdated(targetJoin);
+		}
+
+		// the other branch will not be executed
+		// first, find the concerned tasks
+		List<InternalTask> elseTasks = new ArrayList<>();
+		// elseTasks.add(targetElse);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)